### PR TITLE
Allow theories to occur in a logic name in any order

### PR DESCRIPTION
**Pull request by daniel-larraz**
It also checks that only one arithmetic theory is provided.

### CODE REVIEW

File: src/theory/logic_info.cpp [hunk 1]
```diff
@@ -433,105 +441,170 @@ void LogicInfo::setLogicString(std::string logicString)
...
+            throwTwoArithmeticTheoriesError(arithmetic_theory, "IRDL");
+          enableIntegers();
+          enableReals();
+          arithOnlyDifference();
+          arithmetic_theory = "IRDL";
+          p += 4;
+        }
+        else if (!strncmp(p, "LIA", 3))
+        {
+          if (*arithmetic_theory != '\0')
```
Comment (ajreynol) [inline src/theory/logic_info.cpp:531]: Our guidelines require brackets `if (...) { ... }` not `if() ...`.

**Review by ajreynol — CHANGES_REQUESTED** [review]
Isn't there a more general issue: is `BVBVBV` a valid logic now?

**Comment by daniel-larraz** [comment]
> Isn't there a more general issue: is `BVBVBV` a valid logic now?

After the change, `BVBVBV` is an accepted logic, which is interpreted as simply `BV`. I was leaning towards being more flexible and allowing any combination that doesn't have an ambiguous interpretation. However, I can modify the code to reject multiple instances of the same logic component if needed.

**Comment by ajreynol** [comment]
I think we should probably reject duplicates, yes.

File: src/theory/logic_info.h [hunk 2]
```diff
@@ -293,6 +293,9 @@ class CVC5_EXPORT LogicInfo
     return *this <= other || *this >= other;
   }
 
+ private:
+  void checkDuplicateTheory(theory::TheoryId theory, const char* id);
```
[the remark this answers is not in the document]
Comment (daniel-larraz) [reply src/theory/logic_info.h:306]: Done

**Review by ajreynol — APPROVED** [review]

**Pull request merged**